Index: tinymce-advanced/trunk/mce/anchor/plugin.js
===================================================================
--- tinymce-advanced/trunk/mce/anchor/plugin.js	(revision 894068)
+++ tinymce-advanced/trunk/mce/anchor/plugin.js	(revision 978363)
@@ -13,9 +13,13 @@
 tinymce.PluginManager.add('anchor', function(editor) {
 	function showDialog() {
-		var selectedNode = editor.selection.getNode();
+		var selectedNode = editor.selection.getNode(), name = '';
+
+		if (selectedNode.tagName == 'A') {
+			name = selectedNode.name || selectedNode.id || '';
+		}
 
 		editor.windowManager.open({
 			title: 'Anchor',
-			body: {type: 'textbox', name: 'name', size: 40, label: 'Name', value: selectedNode.name || selectedNode.id},
+			body: {type: 'textbox', name: 'name', size: 40, label: 'Name', value: name},
 			onsubmit: function(e) {
 				editor.execCommand('mceInsertContent', false, editor.dom.createHTML('a', {
